File: doc-examples/example-java/src/test/java/example/TextFunctions.java
package example;

import io.micronaut.el.annotation.ELFunction;

public final class TextFunctions {

    private TextFunctions() {
    }

    @ELFunction // <1>
    public static String shout(String text) {
        return text.toUpperCase() + "!";
    }

    @ELFunction("initials") // <2>
    public static String initialsOf(String text) {
        StringBuilder initials = new StringBuilder();
        for (String word : text.split(" ")) {
            initials.append(word.charAt(0));
        }
        return initials.toString();
    }
}

File: doc-examples/example-python/src/test/python/example/TextFunctions.py
from micronaut.el.annotation import ELFunction


class TextFunctions:

    @staticmethod
    @ELFunction  # <1>
    def shout(text: str) -> str:
        return text.upper() + "!"

    @staticmethod
    @ELFunction("initials")  # <2>
    def initialsOf(text: str) -> str:
        return "".join(word[0] for word in text.split(" "))
